pkg/healthcheck/healthcheck.go

```diff
@@ -1855,6 +1869,47 @@ func (hc *HealthChecker) CheckNamespace(ctx context.Context, namespace string, s
 	return nil
 }
 
+func (hc *HealthChecker) checkClusterNetworks(ctx context.Context) error {
+	nodes, err := hc.kubeAPI.GetNodes(ctx)
+	if err != nil {
+		return err
+	}
+	var podCIDRs []string
+	for _, node := range nodes {
+		podCIDRs = append(podCIDRs, node.Spec.PodCIDR)
+	}
+	clusterNetworks := strings.Split(hc.linkerdConfig.ClusterNetworks, ",")
+	clusterIPNets := make([]*net.IPNet, len(clusterNetworks))
+	for i, clusterNetwork := range clusterNetworks {
+		_, ipNet, err := net.ParseCIDR(clusterNetwork)
+		if err != nil {
+			return err
+		}
+		clusterIPNets[i] = ipNet
+	}
+	var badPodCIDRS []string
+	for _, podCIDR := range podCIDRs {
+		podIP, _, err := net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+		exists := false
+		for _, clusterIPNet := range clusterIPNets {
+			if clusterIPNet.Contains(podIP) {
```

Comment:
Could you give an example if possible? Do you mean a check for something like a `podCIDR = 192.168.0.1/8` and `clusterIPNet = 192.168.0.0/24`?

Reply:
Yup, exactly.  In that case there are IPs in the podCIDR which are not in the cluster IP network.

Reply:
The current logic does cover the above case as far as I understand.
https://play.golang.org/p/B64cxf8GnP9

Reply:
I think you have a typo in your testcase there where you have the second octect of the podCIDR being 162 instead of 168.

https://play.golang.org/p/drV4ETdN7TJ